libs/wingsdk/src/target-tf-aws/bucket.ts

```diff
@@ -103,6 +104,14 @@ export class Bucket extends cloud.BucketBase {
     }
   }
 
+  public addObject(key: string, body: string): void {
+    new S3Object(this, `S3Object-${key}`, {
+      bucket: this.bucket.bucket,
+      key,
```

Comment:
@Chriscbr Restrictions on object names in S3 are quite broad, but there are characters that need special treatment and others to avoid as per this link

https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html

Perhaps we can ignore these limitations for the moment (I imagine very few people want to create a file with @ or $) or we can change the resource-names so that it is not limited to resources but to everything related to names

And I imagine that this can also affect the logical id of the object

Reply:
Logical IDs don't have any hard restrictions on characters or length AFAIK.

Based on the docs you linked, maybe we could offer some warnings if users choose file names with special characters, but I think it's OK to keep things simple for now. For the size of the string, I think every cloud provider has some restrictions on user input sizes, but since the key name is entirely user-provided, I think it's OK to let the cloud provider throw an error instead of handling it in the Wing SDK.